File: rot13.py
# function transformToRot13(stringToTransform) {
# }

# toRot13('c'); // HELLO WORLD
# toRot13('BCRAPYNFFEBBZF'); // OPENCLASSROOMS
# toRot13('PRPV RFG ZBA PBQR FRPERG'); // CECI EST MON CODE SECRET


def transformToRot13(encrypted_word: str):
    alphabet = [
        "a",
        "b",
        "c",
        "d",
        "e",
        "f",
        "g",
        "h",
        "i",
        "j",
        "k",
        "l",
        "m",
        "n",
        "o",
        "p",
        "q",
        "r",
        "s",
        "t",
        "u",
        "v",
        "w",
        "x",
        "y",
        "z",
    ]
    word = ""

    for letter in encrypted_word.lower():
        if letter == " ":
            word += " "
            continue
        word += alphabet[alphabet.index(letter) - 13]

    print(word)

    return word.upper()


assert transformToRot13("URYYB JBEYQ") == "HELLO WORLD"
assert transformToRot13("BCRAPYNFFEBBZF") == "OPENCLASSROOMS"
assert transformToRot13("PRPV RFG ZBA PBQR FRPERG") == "CECI EST MON CODE SECRET"

File: rot13.js
function transformToRot13(encrypted_word) {
  const alphabet = [
    "a",
    "b",
    "c",
    "d",
    "e",
    "f",
    "g",
    "h",
    "i",
    "j",
    "k",
    "l",
    "m",
    "n",
    "o",
    "p",
    "q",
    "r",
    "s",
    "t",
    "u",
    "v",
    "w",
    "x",
    "y",
    "z",
  ];
  let word = "";

  for (let i = 0; i < encrypted_word.length; i++) {
    let letter = encrypted_word[i];
    if (letter === " ") {
      word += " ";
      continue;
    }
    let index = alphabet.indexOf(letter.toLowerCase()) - 13;
    if (index < 0) {
      index = alphabet.length + index;
    }
    word += alphabet[index];
  }

  console.log(word);

  return word.toUpperCase();
}

console.log(transformToRot13("URYYB JBEYQ") === "HELLO WORLD");
console.log(transformToRot13("BCRAPYNFFEBBZF") === "OPENCLASSROOMS");
console.log(transformToRot13("PRPV RFG ZBA PBQR FRPERG") === "CECI EST MON CODE SECRET");
